src/main/java/bio/terra/service/snapshotbuilder/SnapshotRequestDao.java

```diff
@@ -0,0 +1,186 @@
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class SnapshotRequestDao {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final SnapshotAccessRequestResponseMapper responseMapper =
+      new SnapshotAccessRequestResponseMapper();
+  private static final String requestIdField = "id";
+  private static final String datasetIdField = "dataset_id";
+  private static final String requestNameField = "request_name";
+  private static final String requestResearchPurposeField = "request_research_purpose";
+  private static final String snapshotBuilderRequestField = "snapshot_builder_request";
+  private static final String userEmailField = "user_email";
+  private static final String createDateField = "create_date";
+  private static final String updateDateField = "update_date";
+  private static final String statusField = "status";
+
+  @Autowired
+  public SnapshotRequestDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  private static class SnapshotAccessRequestResponseMapper
+      implements RowMapper<SnapshotAccessRequestResponse> {
+    public SnapshotAccessRequestResponse mapRow(ResultSet rs, int rowNum) throws SQLException {
+      return new SnapshotAccessRequestResponse()
+          .id(rs.getObject(requestIdField, UUID.class))
+          .datasetId(rs.getObject(datasetIdField, UUID.class))
+          .requestName(rs.getString(requestNameField))
+          .requestResearchPurpose(rs.getString(requestResearchPurposeField))
+          .request(mapRequestFromJson(rs.getString(snapshotBuilderRequestField)))
+          .createDate(getInstantString(rs, createDateField))
+          .updateDate(getInstantString(rs, updateDateField))
+          .userEmail(rs.getString(userEmailField))
+          .status(SnapshotAccessRequestResponse.StatusEnum.valueOf(rs.getString(statusField)));
+    }
+
+    private String getInstantString(ResultSet rs, String columnLabel) throws SQLException {
+      Timestamp timestamp = rs.getTimestamp(columnLabel);
+      if (timestamp != null) {
+        return timestamp.toInstant().toString();
+      }
+      return null;
+    }
+
+    private static SnapshotBuilderRequest mapRequestFromJson(String json) {
+      if (Objects.nonNull(json)) {
```

Comment:
In general APIs like `Objects.nonNull()` and `Objects.null()` are intended to be used in java streaming operations. (If you're not familiar with the Java streaming operators, you could start here: https://docs.oracle.com/javase/tutorial/collections/streams/index.html. It's old but the basic concepts are all still true, and most APIs used there are still used today. 

Here you can compare against `null` directly
```suggestion
      if (json != null) {
```

Although can this field ever be `null`? I think it's a required field in the database.